### DISCUSSION TITLE
Laravel volt-like syntax for plain blade components

### DISCUSSION THREAD

**Discussion by Wulfheart** [laravel/framework#48537] [Ideas]
Hello,

I really like Laravel Volt and its way to declare components and markup directly together. What I am missing now is a way to also apply this syntax to plain Blade components. It would be nice to also have this option. 

So my idea would be to add it to anonymous components. If you are generally interested I can 
1. write a RFC-like proposal so that the idea can be evaluated further if requested 
2. also contribute this feature

Thank you in advance for your consideration.

**Comment by wizzymore** [comment]
What is stopping you from just using a `<?php ?>` tag at the top of the component and doing stuff like in volt there?

**Reply by Wulfheart** [reply]
I want to have typesafety for the passed parameters.

**Reply by wizzymore** [reply]
You got me hooked on this one, would be really nice to have type safety without the component classes.

**Reply by Wulfheart** [reply]
I think I am going to create a package as POC, a fork from volt and I am going to call it Ampere. 😂

**Reply by luiz-gustavo-agostinho** [reply]
I'm looking for this feature, is there a way to achieve this in 2024?  Thanks!

**Reply by luiz-gustavo-agostinho** [reply]
I think I kinda achieve this by doing:

```php
<?php

$ampere = new class {
    public string $hello = '123';
}; ?>

{{ $ampere->hello }}
```

This works, obviously `$ampere` can be any var name. Not sure if this is what @wizzymore meant with "doing stuff like volt"
